tee-worker/omni-executor/executor-primitives/src/auth.rs

```diff
@@ -1,36 +1,217 @@
 
 pub type VerificationCode = String;
+type Email = String;
+type OmniAccount = String;
+type JwtToken = String;
+
+/// A serializable representation of Identity for JSON interchange.
+/// ```json
+/// {
+///  "type": "Email",
+///  "data": "test@test.com"
+/// }
+/// ```
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(tag = "type", content = "data")]
+pub enum IdentitySerde {
+	Twitter(String),
+	Discord(String),
+	Github(String),
+	Substrate(String), // hex-encoded
+	Evm(String),       // hex-encoded
+	Bitcoin(String),   // hex-encoded
+	Solana(String),    // base58-encoded
+	Email(String),
+	Google(String),
+	Pumpx(String),
+}
+
+impl TryFrom<IdentitySerde> for Identity {
+	type Error = &'static str;
+
+	fn try_from(value: IdentitySerde) -> Result<Self, Self::Error> {
+		match value {
+			IdentitySerde::Twitter(handle) => {
+				Ok(Identity::Twitter(IdentityString::new(handle.as_bytes().to_vec())))
+			},
+			IdentitySerde::Discord(handle) => {
+				Ok(Identity::Discord(IdentityString::new(handle.as_bytes().to_vec())))
+			},
+			IdentitySerde::Github(handle) => {
+				Ok(Identity::Github(IdentityString::new(handle.as_bytes().to_vec())))
+			},
+			IdentitySerde::Substrate(hex_address) => {
+				let bytes = decode_hex(&hex_address).map_err(|_| "Invalid hex encoding")?;
+				let address =
+					Address32::try_from(bytes.as_slice()).map_err(|_| "Invalid address")?;
+				Ok(Identity::Substrate(address))
+			},
+			IdentitySerde::Evm(hex_address) => {
+				let bytes = decode_hex(&hex_address).map_err(|_| "Invalid hex encoding")?;
+				let address =
+					Address20::try_from(bytes.as_slice()).map_err(|_| "Invalid address")?;
+				Ok(Identity::Evm(address))
+			},
+			IdentitySerde::Bitcoin(hex_address) => {
+				let bytes = decode_hex(&hex_address).map_err(|_| "Invalid hex encoding")?;
+				let address =
+					Address33::try_from(bytes.as_slice()).map_err(|_| "Invalid address")?;
+				Ok(Identity::Bitcoin(address))
+			},
+			IdentitySerde::Solana(base58_address) => {
+				let address: Address32 = base58_address
+					.from_base58()
+					.map_err(|_| "Invalid base58 encoding")?
+					.as_slice()
+					.try_into()
+					.map_err(|_| "Invalid address")?;
+				Ok(Identity::Solana(address))
+			},
+			IdentitySerde::Email(handle) => {
+				Ok(Identity::Email(IdentityString::new(handle.as_bytes().to_vec())))
+			},
+			IdentitySerde::Google(handle) => {
+				Ok(Identity::Google(IdentityString::new(handle.as_bytes().to_vec())))
+			},
+			IdentitySerde::Pumpx(handle) => {
+				Ok(Identity::Pumpx(IdentityString::new(handle.as_bytes().to_vec())))
+			},
+		}
+	}
+}
+
+/// A serializable representation of HeimaMultiSignature for JSON interchange.
+/// ```json
+/// {
+///   "type": "Ed25519",
+///   "data": "base64-encoded-signature"
+/// }
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(tag = "type", content = "data")]
+pub enum HeimaMultiSignatureSerde {
+	Ed25519(String),  // base64-encoded
```

Comment:
I would suggest keeping hex encoding here.
These are short length byte arrays and I see no benefit from using base64, do you?